flow/JsonString.h

```diff
@@ -0,0 +1,142 @@
+
+#ifndef JSONSTRING_H
+#define JSONSTRING_H
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+class JsonString;
+class JsonStringArray;
+class JsonStringSetter;
+
+class JsonString {
+	public:
```

Comment:
Since a user defined copy constructor exists, no default move constructor will be generated by the compiler.  The default can be generated with  `JsonString(JsonString &&) = default;`